presto-hive/src/main/java/com/facebook/presto/hive/util/InternalHiveSplitFactory.java

```diff
@@ -184,7 +188,13 @@ public String getPartitionName()
                 splittable,
                 forceLocalScheduling && allBlocksHaveRealAddress(blocks),
                 columnCoercions,
-                bucketConversion));
+                bucketConversion,
+                checkS3SelectPushdownBasedOnCompressionCodec(s3SelectPushdownEnabled, inputFormat, path)));
```

Comment:
Refactored, although statically imported  `isCompressionCodecSupported` to keep it consistent.

Reply:
I suggested leaving the class name so that it's clear this is for S3 Select. The name `isCompressionCodecSupported` by itself doesn't tell the reader that.

Reply:
Ok. As per the feedback on previous revisions my understanding was that static imports is a hard requirement. Thus I created https://github.com/airlift/airbase/issues/168.

If this is not a hard requirement then we should resolve this issue with direction about usage. 

Refactoring.

Reply:
@nezihyigitbasi, requesting your inputs as well regarding this change.

Reply:
Static imports are definitely not a requirement!

Methods **should** be static imported when the name is self-descriptive -- where the class name does not add value and thus detracts from readability. For example, `Math.toIntExact` or `Objects.requireNonNull`.

Methods **should not** be static imported when the name is ambiguous. For example, `ImmutableList.of` or `String.valueOf`.

I haven't seen the original feedback you are referring to, but I'm guessing it was meant to be specifically about `Arrays.asList`. Note that in general, we would prefer `ImmutableList.of` or `ImmutableList.copyOf` and only use `asList` in specific situations like needing nulls (Guava collections don't allow nulls).

Reply:
Sure makes sense. Here is the relevant discussion thread, https://github.com/prestodb/presto/pull/11033#discussion_r234839813.

Reply:
@sameer2086 @electrum summarized it pretty well, static imports is not a hard requirement, and its usage depends on whether there is ambiguity or not. In this particular comment you linked to we recommended static importing a constant (`S3_SELECT_PUSHDOWN_ENABLED`) and there is no ambiguity there.

Reply:
Thanks for the feedback @nezihyigitbasi, closing https://github.com/airlift/airbase/issues/168.